### PR TITLE
Fix 1d recovery and 2d atmosphere fixing

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/PrimitiveFromConservative.cpp [hunk 1]
```diff
@@ -240,8 +242,17 @@ bool PrimitiveFromConservative<OrderedListOfPrimitiveRecoverySchemes,
     }
   }
   if constexpr (ThermodynamicDim == 1) {
+    // Since the primitive recovery scheme is not restricted to lie on the
+    // EOS-satisfying sub-manifold, we project back to the sub-manifold by
+    // recomputing the specific internal energy and specific enthalpyfrom the
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/PrimitiveFromConservative.cpp]: add space after enthalpy

File: tests/Unit/Evolution/VariableFixing/Test_FixToAtmosphere.cpp [hunk 2]
```diff
@@ -82,53 +85,66 @@ template <size_t Dim>
...
 
   CHECK_ITERABLE_APPROX(density, expected_density);
   CHECK_ITERABLE_APPROX(pressure, expected_pressure);
-  CHECK_ITERABLE_APPROX(specific_enthalpy, expected_specific_enthalpy);
+  // CHECK_ITERABLE_APPROX(specific_enthalpy, expected_specific_enthalpy);
   CHECK_ITERABLE_APPROX(specific_internal_energy,
                         expected_specific_internal_energy);
-  CHECK_ITERABLE_APPROX(lorentz_factor, expected_lorentz_factor);
-  CHECK_ITERABLE_APPROX(spatial_velocity, expected_spatial_velocity);
+  // CHECK_ITERABLE_APPROX(lorentz_factor, expected_lorentz_factor);
```
Comment (kidder) [inline tests/Unit/Evolution/VariableFixing/Test_FixToAtmosphere.cpp]: ?

File: tests/Unit/Evolution/VariableFixing/Test_FixToAtmosphere.cpp [hunk 3]
```diff
@@ -82,53 +85,66 @@ template <size_t Dim>
...
+  // CHECK_ITERABLE_APPROX(spatial_velocity, expected_spatial_velocity);
```
Comment (kidder) [inline tests/Unit/Evolution/VariableFixing/Test_FixToAtmosphere.cpp]: ?

File: tests/Unit/Evolution/VariableFixing/Test_FixToAtmosphere.cpp [hunk 4]
```diff
@@ -82,53 +85,66 @@ template <size_t Dim>
...
   //                 (trans_rho - rho_cut) / |v^i|
   expected_spatial_velocity.get(0)[2] = 0.6 * (4.e-12 - 3.e-12) /
                                         (1.e-11 - 3.e-12) * 1.e-4 /
-                                        sqrt(0.6 * 0.6 * 2.0);
+                                        sqrt(0.6 * 0.6 * 2.);
 
   CHECK_ITERABLE_APPROX(density, expected_density);
   CHECK_ITERABLE_APPROX(pressure, expected_pressure);
-  CHECK_ITERABLE_APPROX(specific_enthalpy, expected_specific_enthalpy);
+  // CHECK_ITERABLE_APPROX(specific_enthalpy, expected_specific_enthalpy);
```
Comment (kidder) [inline tests/Unit/Evolution/VariableFixing/Test_FixToAtmosphere.cpp]: uh... ?

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Done and squashed... Thanks for the review :)

**Review by kidder — APPROVED** [review]

**Pull request merged**